internal/utils/utils.go

```diff
@@ -18,34 +18,53 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-func GetScmProviderFromGitRepository(ctx context.Context, k8sClient client.Client, repositoryRef *promoterv1alpha1.GitRepository, obj metav1.Object) (*promoterv1alpha1.ScmProvider, error) {
+func GetScmProviderFromGitRepository(ctx context.Context, k8sClient client.Client, repositoryRef *promoterv1alpha1.GitRepository, obj metav1.Object) (promoterv1alpha1.GenericScmProvider, error) {
 	logger := log.FromContext(ctx)
 
-	var scmProvider promoterv1alpha1.ScmProvider
-	namespace := obj.GetNamespace()
-
-	objectKey := client.ObjectKey{
-		Namespace: namespace,
-		Name:      repositoryRef.Spec.ScmProviderRef.Name,
+	var provider promoterv1alpha1.GenericScmProvider
+	kind := repositoryRef.Spec.ScmProviderRef.Kind
+	if kind == "" {
```

Comment:
Safe to drop this check? Should be enforced by k8s validation. Worst-case scenario we fall back to the error case.